### DISCUSSION TITLE
Where do we stand with Turbomodules?

### DISCUSSION THREAD

**Discussion by shamilovtim** [callstack/react-native-builder-bob#298] [General]
Are Turbomodules on the roadmap for this project or should we lean on [react-native-turbo-starter](https://github.com/talknagish/react-native-turbo-starter) indefinitely? My understanding is that react-native-turbo-starter is not backwards compatible with NativeModules but it's not clear which starter project to contribute to at the moment. To add to the confusion I've heard that Swift support will need to be deprecated for Turbomodules so things are a bit confusing right now for this library.

**Comment by satya164** [comment]
We just added experimental turbo module support with both backward compatibility and without. Currently, only Obj-C and Java are supported for Turbo modules, and Kotlin support will follow.

If you have time, please test it and lemme know if there are any problems or things to be fixed.

**Comment by shamilovtim** [comment]
@satya164 I will test ASAP. Thank you for all that you do.

**Comment by jacobarvidsson** [comment]
@satya164 is this suppose to work for fabric components as well? When I switch the turbo module to a fabric component the build stops working with:
```
Undefined symbols for architecture arm64:
  "nameOfComponent", referenced from:
      _RCTThirdPartyFabricComponentsProvider in libReact-Codegen.a(RCTThirdPartyFabricComponentsProvider.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by satya164** [comment]
I've only tested with turbo modules for now

**Comment by satya164** [comment]
Fabric progress: #273
